tests/unittests/test_refresh_access_token.py

```diff
@@ -0,0 +1,153 @@
+import unittest
+from unittest.mock import MagicMock, patch
+from datetime import datetime, timedelta
+from singer import utils
+from tap_square.client import require_new_access_token, SquareClient
+
+REFRESH_TOKEN_BEFORE = 22
+
+
+class TestRequireNewAccessToken(unittest.TestCase):
+    def setUp(self):
+        self.client = MagicMock()
+        self.access_token = "test_access_token"
+
+    def test_no_access_token(self):
+        """Return true when no access token is provided."""
+        result = require_new_access_token(None, self.client)
+        self.assertTrue(result)
+
+    @patch("singer.http_request_timer")
+    def test_valid_access_token(self, mock_timer):
+        """Return false when the token generated in less than 7 days."""
+        token_expiry = utils.strftime(utils.now() + timedelta(days=26))
+        self.client.o_auth.retrieve_token_status.return_value = MagicMock(
+            is_error=MagicMock(return_value=False),
+            body={"expires_at": token_expiry},
+        )
+        result = require_new_access_token(self.access_token, self.client)
+        self.assertFalse(result)
+
+    @patch("singer.http_request_timer")
+    def test_expired_access_token(self, mock_timer):
+        """Return true when the token is expired."""
+        token_expiry = utils.strftime(utils.now() + timedelta(days=20))
+        self.client.o_auth.retrieve_token_status.return_value = MagicMock(
+            is_error=MagicMock(return_value=False),
+            body={"expires_at": token_expiry},
+        )
+        result = require_new_access_token(self.access_token, self.client)
+        self.assertTrue(result)
+
+    @patch("singer.http_request_timer")
+    def test_almost_expired_access_token(self, mock_timer):
+        """Return false when the token is exact just 7 days ago"""
```

Comment:
Is there some typo in this doc string?
```suggestion
        """Return false when the token is created exactly 7 days ago"""
```

Reply:
I have updated comment